### PR TITLE
Surface Challenges screen fetch failures instead of swallowing them (#101)

### LINKED ISSUE #101: Challenges screen: surface fetch failures instead of swallowing them into the empty/zero state
## Context

Follow-up from PR #100 (issue #98). While wiring the Statistics screen to real data, a review noted that `Statistics.svelte`'s `onMount` swallowed a fetch failure into `[]`, which rendered the new-player "No statistics yet" empty state on a transient network error. PR #100 fixed that screen by distinguishing **error** from **genuine empty success** (surfacing a `toastError` + a dedicated error state, only showing the empty state on an actual empty result).

`Challenges.svelte` still has the identical pattern and should get the same treatment for consistency:

```ts
// UI/new-svelte/src/routes/game/screens/challenges/Challenges.svelte
onMount(async () => {
	try {
		view.playerChallenges = (await ApiRequest.get('Challenges/Player')) ?? [];
	} catch {
		view.playerChallenges = []; // a failed load looks identical to "no progress yet"
	}
	view.loading = false;
});
```

A failed `Challenges/Player` fetch currently presents as a player with zero challenge progress rather than as an error.

## Proposed work

- Mirror the Statistics fix: on a fetch failure, surface a `toastError` (per the toast-system guidance in `docs/frontend.md` that load failures should be reported rather than silently swallowed) and distinguish the error case from a genuine empty result rather than collapsing both into the same UI.
- Keep it consistent with the now-established `Statistics.svelte` convention so the two screens behave the same way.
- Add/adjust the component test to cover the fetch-rejection path (error surfaced, not the empty/zero state).

## Acceptance

- A failed `Challenges/Player` load shows an error affordance (toast + appropriate state), not a misleading "no progress" view.
- A genuine empty result still renders normally.
- Behavior matches `Statistics.svelte`.

## Notes

Scope is intentionally just `Challenges.svelte` — a sweep of the other game screens (`attributes`, `attribute-breakdown`, `inventory`, `options`, `stats`) found no other screen with the error→empty swallow pattern after PR #100.

### PR DESCRIPTION
**Pull request by ginderjeremiah**
## Summary

Implements **#101**. `Challenges.svelte`'s `onMount` previously swallowed a failed `Challenges/Player` fetch into `[]`:

```ts
try {
	view.playerChallenges = (await ApiRequest.get('Challenges/Player')) ?? [];
} catch {
	view.playerChallenges = []; // a failed load looks identical to "no progress yet"
}
```

Because the challenge catalogue itself comes from `staticData`, an empty `playerChallenges` is a *valid* render (every challenge at zero progress) — so a transient network failure was indistinguishable from a genuine new-player no-progress state. This mirrors the exact pattern fixed for `Statistics.svelte` in PR #100, and #101 asks for the same treatment here.

## How it addresses the issue

- **`ChallengesView` now carries an `error` flag** (`challenges-view.svelte.ts`), kept distinct from a genuine no-progress result.
- **On a fetch rejection** `Challenges.svelte` sets `error = true` and surfaces a `toastError`, per the toast-system guidance in `docs/frontend.md` that load failures should be reported rather than silently swallowed.
- **The template renders a dedicated error state** (`data-testid="challenges-error"`) in place of the normal type-rail/detail body when `error` is set, instead of the misleading wall of zero-progress cards.
- **A genuine empty result still renders normally** — the "no progress yet" view is unchanged.

Behavior now matches the established `Statistics.svelte` convention (empty vs. error kept distinct).

## Test plan

- [x] `npm run check` — 0 errors / 0 warnings
- [x] `npm run lint` — clean
- [x] `npm run test:unit:ci` — **538 passing** (66 files), including a new `Challenges.test.ts` component test covering:
  - the happy path (fetches `Challenges/Player`, renders the overview body, no error, no toast)
  - a genuine empty/no-progress result (renders normally, no toast)
  - the fetch-rejection path (error state shown, normal body absent, `toastError` called once)

## Docs

Added an "empty-vs-error kept distinct" note to the Challenge screen section of `docs/frontend.md`, alongside the analogous Statistics-screen note.

## Notes

A repo-wide sweep wasn't needed — the issue notes (and PR #100 confirmed) that after the Statistics fix, no other game screen still has the error→empty swallow pattern. Scope is intentionally just `Challenges.svelte`.

Closes #101

https://claude.ai/code/session_01WjCibZGTrAtvGgB4WR1LPz

---
_Generated by [Claude Code](https://claude.ai/code/session_01WjCibZGTrAtvGgB4WR1LPz)_

### CODE REVIEW

File: UI/new-svelte/src/tests/routes/game/screens/challenges/Challenges.test.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+	staticData.enemies = [];
+	staticData.zones = [];
+	staticData.skills = [];
+	staticData.challengeTypes = [
+		{ id: EChallengeType.EnemiesKilled, goalComparison: EChallengeGoalComparison.AtLeast, name: 'Enemies Killed' }
+	];
+	staticData.challenges = [
+		challenge({ id: 1, name: 'First Blood', challengeTypeId: EChallengeType.EnemiesKilled, progressGoal: 10 })
+	];
+	mockToastError.mockClear();
```
Comment (ginderjeremiah) [inline UI/new-svelte/src/tests/routes/game/screens/challenges/Challenges.test.ts:54]: `mockGet` isn't cleared between tests — `beforeEach` resets its resolved value (`.mockResolvedValue(...)`) but leaves the call history accumulating. The only assertion that touches it (`toHaveBeenCalledWith`) doesn't care about call count so there's no current failure, but it's inconsistent with `mockToastError.mockClear()` and could silently mask a later `toHaveBeenCalledTimes` assertion. Add `mockGet.mockClear()` here alongside the `mockToastError` clear.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by ginderjeremiah — COMMENTED** [review]
## Review: Surface Challenges fetch failures instead of swallowing them (#101)

**Intended verdict:** Approve — no blocking findings. (Submitting as `COMMENT` because I'm the PR author.)

### What's good

- The fix exactly mirrors the established `Statistics.svelte` convention — same `error` flag on the view-model, same `toastError` on rejection, same dedicated error state in the template, same guard that a genuine empty result still renders normally. The parallel is precise and the two screens now behave consistently.
- `ChallengesView.error` is the right home for this state: a plain `$state` field on the reactive view-model keeps the error representation with the rest of the fetch-driven state and leaves the component logic minimal.
- The error UI uses only `var(--text-primary)` and `var(--text-tertiary)` — no hard-coded colours — following the theming rules.
- Three tests cover all three acceptance criteria from the issue (happy path, genuine empty, fetch rejection) and are concise and readable.
- The doc addition is accurate and well-placed; no over-documentation.

### Nits

- **`Challenges.test.ts:54`** — `mockGet` isn't cleared between tests. See inline comment.

### Test coverage

All three acceptance-criteria paths are covered and the mocking approach (selective override of `$lib/api` and `$stores`, keeping the real `registerTooltipComponent` and other stores live) mirrors the Statistics screen test pattern. Coverage is sufficient.


---
_Generated by [Claude Code](https://claude.ai/code)_
